### ISSUE TITLE
Removing print statements

### ISSUE THREAD

**Issue by jixishan** [HPCE/hpce-2018-cw6#16]
Are we allowed to remove print statements from the code to improve speed?

**Comment by jjd06** [comment]
Yes (so long as they aren't vital to functionality), but do remember that making things faster isn't the purpose of this exercise. In your report, you could propose the removal of `printf`s, but, unless you can show that doing so would give noticeable speedup, that would probably sound a bit silly.
